### PR TITLE
[telemetry] rename memorylimiter metrics to include memory_limiter infix

### LINKED ISSUE #11203: decision around deprecated metrics
The following metrics are being deprecated as they were only used in the memory limiter  processor:
    - `otelcol_processor_accepted_log_records`
    - `otelcol_processor_accepted_metric_points`
    - `otelcol_processor_accepted_spans`
    - `otelcol_processor_dropped_log_records`
    - `otelcol_processor_dropped_metric_points`
    - `otelcol_processor_dropped_spans`
    - `otelcol_processor_refused_log_records`
    - `otelcol_processor_refused_metric_points`
    - `otelcol_processor_refused_spans`

This issue is to make a decision as to what to do with these metrics. They could be renamed to `otelcol_processor_memory_limiter_*` to identify that these metrics are specific to this one processor.

### PR DESCRIPTION
**Pull request by JamieDanielson**
#### Description

Deprecated metrics used exclusively in memory limiter processor now have an infix of `memory_limiter`. For example, `otelcol_processor_accepted_spans` is now `otelcol_processor_memory_limiter_accepted_spans`.

~Some tests also got formatting updates during the `make generate` and `make mdatagen-test` steps, happy to remove them if preferred.~
Edit: 🤔  Actually that seems to fail tests, so I've reverted those changes. I saw the suggestion to run `mdatagen-test` in the `mdatagen` README, but maybe that should be updated (and/or changes made to prevent the formatting change?)

#### Link to tracking issue

- Fixes #11203 

#### Documentation

Auto-generated documentation has been updated

### CODE REVIEW

File: processor/processorhelper/metadata.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ telemetry:
         value_type: int
         monotonic: true
 
-    processor_accepted_spans:
+    processor_memory_limiter_accepted_spans:
```
Comment (codeboten) [inline processor/processorhelper/metadata.yaml:33]: These metrics will need to be moved to the memorylimiter's metadata configuration https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/memorylimiterprocessor/metadata.yaml

With the existing metrics staying deprecated for another few versions in the processorhelper

Comment (JamieDanielson) [reply processor/processorhelper/metadata.yaml:33]: Ah yep makes sense. I'm going to put this back to draft to make updates

**Comment by JamieDanielson** [comment]
Sorry y'all this fell off my radar, will pick it back up this week.

**Comment by msaicharan** [comment]
@JamieDanielson  :   any updates wrt to these changes ?

**Pull request closed without merging**